doc/source/WorkingWithThePythonScriptingAPI/ThePythonEditor/index.md

```diff
@@ -0,0 +1,72 @@
+```eval_rst
+.. role:: raw-html(raw)
+    :format: html
+```
+
+# The Python Editor #
+
+The _Python Editor_ is a type of editor in the main Gaffer application that lets you build and modify the node graph, test expressions, and inspect data by inputting and executing Python code. In the default layout, it resides in the bottom-right panel, in a tab next to the _Hierarchy View_.
+
+The _Python Editor_ is split into two areas. The bottom-half is the input field. The top-half is the output. The input field of the _Python Editor_ functions like a basic plain text editor. You can type, select, cut, copy, paste, and execute code. 
+
+![A Python Editor with its input and the output field](images/interfacePythonEditorOutput.png "A Python Editor with its input and the output field")
+
+The output field behaves like a typical console output. It records the output stream, including return values and errors. Errors appear in red.
+
+![Errors in a Python Editor's output field](images/interfacePythonEditorError.png "Errors in a Python Editor's output field")
+
+> Tip :
+> When creating a new node using the _Python Editor_, you must first `import` the node's module, which will persist for the lifetime of the editor. However, you never need to `import` the [Gaffer](../../Reference/NodeReference/Gaffer/index.md) or [IECore](../../Reference/NodeReference/IECore/index.md) modules. They are loaded by default.
+
+
+## Instructions ##
+
+
+### Executing Code ###
+
+To execute code with the _Python Editor:_
+
+1. Enter code into the input field.
+2. Highlight any portion of the input you wish to preserve for reuse.
+3. Hit <kbd>Control</kbd> + <kbd>Enter</kbd>.
+
+> Tip :
+> Highlighting code to preserve it for reuse can be a very useful debugging aid.
+
+
+### Grabbing data from other editors ###
+
+The _Python Editor_ can refer to the same nodes, plugs, objects, scene locations, and inspected scene data accessible from the various other editors. However, this requires inputting precise dictionary syntax, types, or code. Instead of referring to these kinds of data manually, you can use a shortcut: simply drag the node, plug, or location from its source editor into the _Python Editor's_ input field.
+
+The specific shortcut for dragging data varies depending on the source editor:
+
+```eval_rst
+======================== ================= ====================== ============
+Data Type                Source Editor     Shortcut               Cursor
```

Comment:
> Data Type

To be accurate, the below aren't (all) data types, they're 'logical things'.   In reality, these are: 

| Old |  Actual type |
|-|-|
| Node | `Gaffer.Node` or `[Gaffer.Node]` |
| Plug | `Gaffer.Plug` |
| Plug value | `float`/`string`/`int`,`bool`/`IECore.Data`, etc... |
| Object(s) | `[str]` |
| Scene locations(s) | `[str]` |
| Scene data | `float`/`string`/`int`,`bool`/`IECore.Data`, etc... |

I wonder if its worth re-arranging so that it lists types you might want against where you get them from? if you don't want to get into the actual types, then maybe change the heading? But I think there is definitely value in knowing that if I drag an object from the viewer I get a list of paths.